nil pointer dereference when using a wrong selector

Defining a `Profile`'s or `ClusterProfile`'s selector with a wrong format, the Addon Controller crashes with the following stacktrace:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x17a1a6e]

goroutine 359 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:116 +0x1e5
panic({0x1e94e80?, 0x36760c0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/projectsveltos/libsveltos/lib/clusterproxy.getMatchingCAPIClusters({0x24ef410, 0xc0017b2c00}, {0x24fe340?, 0xc0005c0900?}, {0x0, 0x0}, {0xc0013023c0, 0x1a}, {{0x24f42e8, 0xc0011fb400}, ...})
        /go/pkg/mod/github.com/projectsveltos/libsveltos@v0.21.0/lib/clusterproxy/cluster_utils.go:421 +0x38e
github.com/projectsveltos/libsveltos/lib/clusterproxy.GetMatchingClusters({0x24ef410, 0xc0017b2c00}, {0x24fe340, 0xc0005c0900}, {0x0, 0x0}, {0xc0013023c0, 0x1a}, {{0x24f42e8, 0xc0011fb400}, ...})
        /go/pkg/mod/github.com/projectsveltos/libsveltos@v0.21.0/lib/clusterproxy/cluster_utils.go:478 +0x85
github.com/projectsveltos/addon-controller/controllers.getMatchingClusters({0x24ef410, 0xc0017b2c00}, {0x24fe340, 0xc0005c0900}, {0xc0013023c0, 0x1a}, 0x21?, {{0x24f42e8, 0xc0011fb400}, 0x0})
        /workspace/controllers/profile_utils.go:57 +0xf1
github.com/projectsveltos/addon-controller/controllers.(*ProfileReconciler).reconcileNormal(0xc000400140, {0x24ef410, 0xc0017b2c00}, 0xc00055be00)
        /workspace/controllers/profile_controller.go:185 +0x1d0
github.com/projectsveltos/addon-controller/controllers.(*ProfileReconciler).Reconcile(0xc000400140, {0x24ef410?, 0xc0017b2c00?}, {{{0xc0013023c0, 0x1a}, {0xc0017aead6, 0x6}}})
        /workspace/controllers/profile_controller.go:150 +0x5bf
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x24f42e8?, {0x24ef410?, 0xc0017b2c00?}, {{{0xc0013023c0?, 0xb?}, {0xc0017aead6?, 0x0?}}})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119 +0xb7
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0006a4280, {0x24ef448, 0xc0005d72c0}, {0x1f78980?, 0xc00103ae60?})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316 +0x3cc
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0006a4280, {0x24ef448, 0xc0005d72c0})
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266 +0x1c9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227 +0x79
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2 in goroutine 259
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:223 +0x565
```

This can be triggered as it follows

```yaml
apiVersion: config.projectsveltos.io/v1alpha1
kind: Profile
metadata:
  name: cilium
  namespace: dtranchitella-1000-default
spec:
  clusterSelector: cni=cilium,  ## please notice the extra comma
  helmCharts:
  - chartName: cilium/cilium
    chartVersion: v1.14.5
    helmChartAction: Install
    releaseName: cilium
    releaseNamespace: kube-system
    repositoryName: cilium
    repositoryURL: https://helm.cilium.io/
  reloader: false
  stopMatchingBehavior: WithdrawPolicies
  syncMode: ContinuousWithDriftDetection
```

The error occurred on the last release (v0.21.0)

--- a/lib/clusterproxy/cluster_utils.go
+++ b/lib/clusterproxy/cluster_utils.go
@@ -18,6 +18,7 @@ package clusterproxy
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"sync/atomic"
 
@@ -43,7 +44,8 @@ import (
 )
 
 const (
-	kubernetesAdmin = "kubernetes-admin"
+	kubernetesAdmin    = "kubernetes-admin"
+	nilSelectorMessage = "selector is nil"
 )
 
 var (
@@ -386,6 +388,11 @@ func GetListOfClustersForShardKey(ctx context.Context, c client.Client, namespac
 func getMatchingCAPIClusters(ctx context.Context, c client.Client, selector labels.Selector,
 	namespace string, logger logr.Logger) ([]corev1.ObjectReference, error) {
 
+	if selector == nil {
+		logger.V(logs.LogInfo).Info(nilSelectorMessage)
+		return nil, fmt.Errorf("%s", nilSelectorMessage)
+	}
+
 	present, err := isCAPIPresent(ctx, c, logger)
 	if err != nil {
 		logger.Error(err, "failed to verify if ClusterAPI Cluster CRD is installed")
@@ -434,6 +441,11 @@ func getMatchingCAPIClusters(ctx context.Context, c client.Client, selector labe
 func getMatchingSveltosClusters(ctx context.Context, c client.Client, selector labels.Selector,
 	namespace string, logger logr.Logger) ([]corev1.ObjectReference, error) {
 
+	if selector == nil {
+		logger.V(logs.LogInfo).Info(nilSelectorMessage)
+		return nil, fmt.Errorf("%s", nilSelectorMessage)
+	}
+
 	listOptions := []client.ListOption{}
 	if namespace != "" {
 		listOptions = append(listOptions, client.InNamespace(namespace))
@@ -472,6 +484,11 @@ func getMatchingSveltosClusters(ctx context.Context, c client.Client, selector l
 // GetMatchingClusters returns all Sveltos/CAPI Clusters currently matching selector
 func GetMatchingClusters(ctx context.Context, c client.Client, selector labels.Selector,
 	namespace string, logger logr.Logger) ([]corev1.ObjectReference, error) {
+
+	if selector == nil {
+		logger.V(logs.LogInfo).Info(nilSelectorMessage)
+		return nil, fmt.Errorf("%s", nilSelectorMessage)
+	}
 
 	matching := make([]corev1.ObjectReference, 0)
 

